cmd/subnetcmd/upgradecmd/generate.go

```diff
@@ -315,6 +324,80 @@ func promptTxAllowListParams(precompiles *[]params.PrecompileUpgrade, date time.
 	return nil
 }
 
+func getCClient(apiEndpoint string, blockchainID string) (ethclient.Client, error) {
+	cClient, err := ethclient.Dial(fmt.Sprintf("%s/ext/bc/%s/rpc", apiEndpoint, blockchainID))
+	if err != nil {
+		return nil, err
+	}
+	return cClient, nil
+}
+
+func ensureAdminsHaveBalanceLocalNetwork(admins []common.Address, blockchainID string) error {
+	cClient, err := getCClient(constants.LocalAPIEndpoint, blockchainID)
+	if err != nil {
+		return err
+	}
+
+	for _, admin := range admins {
+		// we can break at the first admin who has a non-zero balance
+		accountBalance, err := getAccountBalance(context.Background(), cClient, admin.String())
+		if err != nil {
+			return err
+		}
+		if accountBalance > float64(0) {
+			return nil
+		}
+	}
+
+	return errors.New("none of the addresses in the transaction allow list precompile have any tokens allocated to them. Currently, no address can transact on the network. Airdrop some funds to one of the allow list addresses to continue")
+}
+
+func ensureAdminsHaveBalance(admins []common.Address, subnetName string) error {
+	if len(admins) < 1 {
+		return nil
+	}
+
+	if !app.GenesisExists(subnetName) {
+		ux.Logger.PrintToUser("The provided subnet name %q does not exist", subnetName)
+		return nil
+	}
+
+	// read in sidecar
+	sc, err := app.LoadSidecar(subnetName)
+	if err != nil {
+		return err
+	}
+	switch sc.VM {
+	case models.SubnetEvm:
+		// Currently only checking if admins have balance for subnets deployed in Local Network
+		if networkData, ok := sc.Networks["Local Network"]; ok {
+			blockchainID := networkData.BlockchainID.String()
+			if err = ensureAdminsHaveBalanceLocalNetwork(admins, blockchainID); err != nil {
+				return err
+			}
+		}
+	default:
+		app.Log.Warn("Unknown genesis format", zap.Any("vm-type", sc.VM))
```

Comment:
I feel this is error text is slightly incorrect. It's an unexpected VM type, in fact.